server/routes/users.js

```diff
@@ -3,167 +3,473 @@ var express = require('express');
+      user_id:  ids[0],
+      company_id: company_id,
+      rank: rank
+    }
+
+    inserts = []
+
+    if(diet_id){
+      inserts.push({
+        values: diet_values,
+        table_name: 'user_diet'
+      });
+    }
+    if(occupation_id){
+      inserts.push({
+        values: occupation_values,
+        table_name: 'user_occupation'
+      });
+    }
+    if(position_id){
+      inserts.push({
+        values: position_values,
+        table_name: 'user_position'
+      });
+    }
+    if(major_id){
+      inserts.push({
+        values: major_values,
+        table_name: 'user_major'
+      });
+    }
+    if(company_id && rank){
+      inserts.push({
+        values: company_rank_values,
+        table_name: 'user_company_rank'
+      });
+    }
+    console.log(ids[0]); 
+    return Promise.map(inserts, function(insert_obj){
+        return knex.insert(insert_obj.values).into(insert_obj.table_name).transacting(trx);
+      });
+    });
+  })
+  .then(result => {
+   res.send(util.message('Successfully inserted new user: ' + req.query.first_name + 'with diet and occupation id'));    
+  })
+  .catch(err => { return next(err) 
+  // If we get here, that means that neither the user insert nor any of the other inserts hav taken place
+  })
+});
+
+
+// Login a user
+router.put('/login', function(req, res, next) {
+  knex('swe_user').where({ email: req.query.email, password: req.query.password })
+    .then(result => {
+      res.send(util.message('Successfully logged in user with email: ' + req.query.email));
+    })
+    .catch(err => { return next(err) });
 });
 
 // GET user info by user_id
 router.get('/:user_id/id', function(req, res, next) {
-  const user_id = req.params.user_id;
-  db.query('SELECT * FROM swe_user WHERE id = $1', [user_id], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('swe_user').select().where({ id: req.params.user_id })
+    .then(result => {
+      if (result.length) {
+        res.json(result);
+      } else {
+        util.throwError(404, 'No users matching user_id = ' + req.params.user_id + ' found');
+      }
+    })
+    .catch(err => { return next(err) });
 });
 
 // GET whether user is admin or not
 router.get('/:user_id/admin', function(req, res, next) {
-  const user_id = req.params.user_id;
-  db.query('SELECT is_admin FROM swe_user WHERE id = $1', [user_id], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('swe_user').select('is_admin').where({ id: req.params.user_id })
+    .then(result => {
+      if (result.length) {
+          res.json(result);
+      } else {
+        util.throwError(404, 'No users matching user_id = ' + req.params.user_id + ' found');
+      }
+    })
+    .catch(err => { return next(err) });
 });
 
-// GET a user's past events-HELP NEEDED
+// Update user info
+var inserts = []
+router.put('/:user_id', function(req, res, next) {
+  var user_id = req.params.user_id;
```

Comment:
Same comments as with the /register request

Reply:
fixed